### PR TITLE
GitHub Actions implementation

### PR DESCRIPTION
**Pull request by ajax-butriy-v**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM amazoncorretto:17-alpine3.17
+WORKDIR /workspace/app
+
+COPY /build/libs/*.jar ./app.jar
+EXPOSE 8080
+CMD ["java", "-jar", "./app.jar"]
```
Comment (ajax-yurkovskyi-t) [inline Dockerfile]: add empty line

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
+      username: ${MONGODB_USER}
+      password: ${MONGODB_PASSWORD}
+      host: ${MONGODB_HOST}
+      port: ${MONGODB_PORT}
+      database: ${MONGODB_DATABASE}
+      authentication-database: admin
 server:
   servlet:
-    context-path: /api/v1
+    context-path: /api/v1
```
Comment (ajax-yurkovskyi-t) [inline src/main/resources/application.yml]: empty line here too

**Review by ajax-yurkovskyi-t — COMMENTED** [review]
Good job

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+delta-coverage = "2.4.0"
+
+[plugins]
+kotlin-jvm = { id = "org.jetbrains.kotlin.jvm", version.ref = "kotlin" }
+kotlin-spring-plugin = { id = "org.jetbrains.kotlin.plugin.spring", version.ref = "kotlin" }
+spring-boot = { id = "org.springframework.boot", version.ref = "spring-boot" }
+spring-dependency-management = { id = "io.spring.dependency-management", version.ref = "spring-dependency-management" }
+detekt = { id = "io.gitlab.arturbosch.detekt", version.ref = "detekt" }
+delta-coverage = { id = "io.github.surpsg.delta-coverage", version.ref = "delta-coverage" }
+
```
Comment (ajax-yurkovskyi-t) [inline gradle/libs.versions.toml:18]: remove redundant line

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+jackson-module-kotlin = { module = "com.fasterxml.jackson.module:jackson-module-kotlin", version.ref = "jackson-module-kotlin" }
+spring-boot-starter-data-mongodb = { module = "org.springframework.boot:spring-boot-starter-data-mongodb", version.ref = "spring-boot" }
+spring-boot-starter-validation = { module = "org.springframework.boot:spring-boot-starter-validation", version.ref = "spring-boot" }
+spring-boot-docker-compose = { module = "org.springframework.boot:spring-boot-docker-compose", version.ref = "spring-boot" }
+spring-boot-starter-test = { module = "org.springframework.boot:spring-boot-starter-test", version.ref = "spring-boot" }
+springmockk = { module = "com.ninja-squad:springmockk", version.ref = "springmockk" }
+kotlin-test-junit5 = { module = "org.jetbrains.kotlin:kotlin-test-junit5", version.ref = "kotlin" }
+junit-platform-launcher = { module = "org.junit.platform:junit-platform-launcher", version.ref = "junit-platform-launcher" }
+detekt-formatting = { module = "io.gitlab.arturbosch.detekt:detekt-formatting", version.ref = "detekt" }
+
```
Comment (ajax-yurkovskyi-t) [inline gradle/libs.versions.toml]: redundant line

File: docker-compose.yaml [hunk 5]
```diff
@@ -1,4 +1,19 @@
...
+    build:
+      context: .
+      dockerfile: Dockerfile
+    environment:
+      APPLICATION_NAME: PassManagingApplication
+      MONGODB_PORT: 27017
+      MONGODB_HOST: mongo
+      MONGODB_DATABASE: test_db
+      MONGODB_USER: root
+      MONGODB_PASSWORD: example
```
Comment (ajax-torbin-v) [inline docker-compose.yaml]: Why you don't use .env file?

File: docker-compose.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
+services:
+  api:
```
Comment (ajax-shulepa-a) [inline docker-compose.yaml]: maybe app?

**Review by ajax-shulepa-a — APPROVED** [review]

File: .github/workflows/gradle-setup.yml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
```
Comment (ajax-samondros-b) [inline .github/workflows/gradle-setup.yml]: use coretto jdk

**Pull request merged**